### PR TITLE
NP-1556 : Expose and update openapi for Update doi request

### PR DESCRIPTION
**Pull request by norrs**
https://unit.atlassian.net/browse/NP-1556

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -201,6 +286,28 @@ Resources:
...
+          TABLE_NAME: !Ref DoiRequestsTableName
+          INDEX_NAME: !Ref DoiRequestsIndexName
+      Policies:
+        - DynamoDBCrudPolicy:
+            TableName: !Ref DoiRequestsTableName
+      Events:
+        GetEvent:
+          Type: Api
+          Properties:
+            Path: /{publicationIdentifier}/
```
Comment (norrs) [inline template.yaml:308]: Trailing slash?

**Review by norrs — COMMENTED** [review]
This API eats `identifier` (uuid) and not `id` (IRI), isn't this fine? The API only know how to handle local resources anyway?

Or should we change to accept `id` as query argument instead of path?

@elkebab @axthosarouris @brinxmat

**Review by axthosarouris — APPROVED** [review]

File: template.yaml [hunk 2]
```diff
@@ -127,10 +127,82 @@ Resources:
...
+                    type: string
+                    format: uuid
+              requestBody:
+                content:
+                  application/json:
+                    schema:
+                      $ref: '#/components/schemas/UpdateDoiRequest'
+              responses:
+                '202':
+                  description: OK
```
Comment (brinxmat) [inline template.yaml:150]: This is not OK

File: template.yaml (hunk 1 above)
Comment (brinxmat) [reply template.yaml:308]: Typically not, according to the docs, the path MUST begin with '/', but it doesn't make sense to have the trailing slash here, does it?

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Comment by norrs** [comment]
Ref. https://unit-norge.slack.com/archives/CN005AQCE/p1601620868041700

@axthosarouris @brinxmat : Do we change this API maybe to eat the IRI? Or is this something the frontend handles?

**Review by axthosarouris — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Comment by axthosarouris** [comment]
> Ref. https://unit-norge.slack.com/archives/CN005AQCE/p1601620868041700
> 
> @axthosarouris @brinxmat : Do we change this API maybe to eat the IRI? Or is this something the frontend handles?

I think right now the publications do not have a URI-ID but only an Identifier. So until this is fixed, I do not think we should expect the frontend to construct a URI from an identifier that we now will split to extract the identifier. When we will have the ID in place for the publications, then we can change the APIs

**Review by brinxmat — APPROVED** [review]

**Comment by norrs** [comment]
Hm, I merged before I got feedback from @elkebab , but we know we will have a new iteration tho when we deal with fixing IRI on the APIs.

**Pull request merged**